internal/table/table.go

```diff
@@ -138,18 +144,29 @@ func equalizeWidths(colWidths []int, percentage int) {
 	}
 }
 
-func rescaleColumns(colWidths []int, targetTotal int) {
+func rescaleColumns(colWidths []int, targetTotal int, vertical bool, verticalHeaderWidth int) {
 	total := float64(mathutils.Total(colWidths...))
 	multiplier := float64(targetTotal) / total
 
+	originalWidths := make([]int, len(colWidths))
 	for n := range colWidths {
+		originalWidths[n] = colWidths[n]
 		colWidths[n] = int(float64(colWidths[n]) * multiplier)
 	}
 
 	// Account for floats that got rounded
 	if len(colWidths) > 0 {
 		colWidths[len(colWidths)-1] += targetTotal - mathutils.Total(colWidths...)
 	}
+
+	// If vertical, respect the header width
+	if vertical && len(colWidths) > 0 && colWidths[0] < verticalHeaderWidth {
```

Comment:
It looks like `verticalHeaderWidth` is either the value of `colWidths[0]` or just 0. So either this conditional doesn't trigger, or it triggers if `colWidths[0] < 0`, which feels inappropriate.

Reply:
As in my other comment, `verticalHeaderWidth` is the value before equalization so it can differ from `colWidths[0]`. I'll add a comment here as well.